analyzer/node_stats/node_stats.go

```diff
@@ -54,22 +61,44 @@ func NewAnalyzer(
 }
 
 func (p *processor) GetItems(ctx context.Context, limit uint64) ([]common.Layer, error) {
-	return []common.Layer{common.LayerConsensus}, nil
+	return []common.Layer{
+		common.LayerConsensus,
+		common.LayerEmerald,
+		common.LayerSapphire,
+	}, nil
 }
 
 func (p *processor) ProcessItem(ctx context.Context, batch *storage.QueryBatch, layer common.Layer) error {
 	p.logger.Debug("fetching node height", "layer", layer)
-	// We currently only support consensus. Update when this is no longer the case.
-	if layer != common.LayerConsensus {
+	// Consensus
+	if layer == common.LayerConsensus {
+		latestBlock, err := p.source.GetBlock(ctx, consensusAPI.HeightLatest)
+		if err != nil {
+			return fmt.Errorf("error fetching latest block height for layer %s, %w", layer, err)
+		}
+		batch.Queue(queries.NodeHeightUpsert,
+			layer,
+			latestBlock.Height,
+		)
+		return nil
+	}
+	// Runtimes
+	var latestBlock *nodeapi.RuntimeBlockHeader
+	var err error
+	switch layer {
+	case common.LayerEmerald:
+		latestBlock, err = p.emeraldSource.GetBlockHeader(ctx, runtimeAPI.RoundLatest)
+	case common.LayerSapphire:
+		latestBlock, err = p.sapphireSource.GetBlockHeader(ctx, runtimeAPI.RoundLatest)
+	default:
 		return fmt.Errorf("unsupported layer %s", layer)
 	}
```

Comment:
Stylistic only: Would it make sense to unify the consensus flow and the runtime flow more? Something like this:

```suggestion
	latestHeight := -1  // will be fetched from the node
	switch layer {
	case common.LayerConsensus:
		latestBlock, err = p.source.GetBlock(ctx, consensusAPI.HeightLatest)
		if err != nil {
			return fmt.Errorf("error fetching latest block height for layer %s: %w", layer, err)
		}
		latestHeight = latestBlock.Height
	case common.LayerEmerald:
		latestBlock, err = p.emeraldSource.GetBlockHeader(ctx, runtimeAPI.RoundLatest)
		if err != nil {
			return fmt.Errorf("error fetching latest block height for layer %s: %w", layer, err)
		}
		latestHeight = latestBlock.Round
	case common.LayerSapphire:
		latestBlock, err = p.sapphireSource.GetBlockHeader(ctx, runtimeAPI.RoundLatest)
		if err != nil {
			return fmt.Errorf("error fetching latest block height for layer %s: %w", layer, err)
		}
		latestHeight = latestBlock.Round		
	default:
		return fmt.Errorf("unsupported layer %s", layer)
	}
```